src/StockServiceConfig.php

```diff
@@ -51,7 +51,7 @@ public function getAvailabilityLocations(Context $context, PurchasableEntityInte
    * {@inheritdoc}
    */
   public function getTransactionLocation(Context $context, PurchasableEntityInterface $entity, $quantity) {
-    $locations = $this->getLocationList($entity);
+    $locations = $this->getAvailabilityLocations($context, $entity);
```

Comment:
good catch. I suspect this is only used by the "always in stock" otherwise would have come up sooner